components/automate-cli/cmd/chef-automate/ha_node_operations_utils.go

```diff
@@ -703,3 +738,170 @@ func readConfigAWS(path string) (AwsConfigToml, error) {
 	}
 	return config, nil
 }
+
+type NodeObject struct {
+	CmdString       string
+	OutputFile      []string
+	InputFile       []string
+	InputFilePrefix string
+	NodeType        string
+}
+
+func NewNodeObjectWithOutputFile(cmdString string, outFile []string, inputFile []string, inputFilePrefix string, nodeType string) *NodeObject {
+	return &NodeObject{cmdString, outFile, inputFile, inputFilePrefix, nodeType}
+}
+
+func (nu *NodeUtilsImpl) parseAndMoveConfigFilteToWorkspaceDir(outFiles []string, outputDirectory string) error {
+
+	err := removeOutputHeaderInConfigFile(outFiles[0])
+	if err != nil {
+		return errors.Wrap(err, "error on removing output header in fetched config")
+	}
+	if err = fileutils.Move(outFiles[0], outputDirectory+outFiles[0]); err != nil {
+		return err
+	}
+	return nil
+}
+
+func removeOutputHeaderInConfigFile(filePath string) error {
+	return fileutils.RemoveFirstLine(filePath)
+}
+
+func getNodeObjectsToFetchConfigFromAllNodeTypes() []*NodeObject {
+	nodeObjects := []*NodeObject{
+		NewNodeObjectWithOutputFile(fmt.Sprintf(GET_FRONTEND_CONFIG, AUTOMATE_TOML), []string{AUTOMATE_TOML}, nil, "", AUTOMATE),
+		NewNodeObjectWithOutputFile(fmt.Sprintf(GET_FRONTEND_CONFIG, CHEF_SERVER_TOML), []string{CHEF_SERVER_TOML}, nil, "", CHEF_SERVER),
+	}
+	if !isManagedServicesOn() {
+		backendNodeObjects := []*NodeObject{
+			NewNodeObjectWithOutputFile(fmt.Sprintf(GET_BACKEND_CONFIG, POSTGRESQL, " > "+POSTGRESQL_TOML), []string{POSTGRESQL_TOML}, nil, "", POSTGRESQL),
+			NewNodeObjectWithOutputFile(fmt.Sprintf(GET_BACKEND_CONFIG, OPENSEARCH, " > "+OPENSEARCH_TOML), []string{OPENSEARCH_TOML}, nil, "", OPENSEARCH),
+		}
+		nodeObjects = append(nodeObjects, backendNodeObjects...)
+	}
+	return nodeObjects
+}
+
+func getNodeObjectsToPatchWorkspaceConfigToAllNodes() []*NodeObject {
+	timestamp := time.Now().Format("20060102150405")
+	fmt.Println("====================================================================")
+	fmt.Println("sync Config to all frontend nodes")
+	frontendPrefix := "frontend" + "_" + timestamp + "_"
+	frontend := fmt.Sprintf(FRONTEND_COMMAND, PATCH, frontendPrefix+AUTOMATE_TOML, DATE_FORMAT)
+	chefserver := fmt.Sprintf(FRONTEND_COMMAND, PATCH, frontendPrefix+CHEF_SERVER_TOML, DATE_FORMAT)
+	nodeObjects := []*NodeObject{
+		NewNodeObjectWithOutputFile(frontend, nil, []string{AUTOMATE_HA_AUTOMATE_NODE_CONFIG_DIR + AUTOMATE_TOML}, frontendPrefix, AUTOMATE),
+		NewNodeObjectWithOutputFile(chefserver, nil, []string{AUTOMATE_HA_AUTOMATE_NODE_CONFIG_DIR + CHEF_SERVER_TOML}, frontendPrefix, CHEF_SERVER),
+	}
+	return nodeObjects
+}
+
+// Execute 'config show' command in specific service and fetch the output file to bastion
+func executeCustomCmdOnEachNodeType(outputFiles []string, inputFiles []string, inputFilesPrefix string, service string, cmdString string, singleNode bool, infra *AutomateHAInfraDetails) error {
```

Comment:
This function is not required please use the cmdutil available in the nodeutil struct. and move this business logic to a common method and call it from your add/remove node

Reply:
Updated the logic to use cmdUtil in nodeUtil.
Moved business logic to the `createNodeMap` function